piptools/cache.py

```diff
@@ -51,6 +51,8 @@ class DependencyCache(object):
     def __init__(self, cache_dir=None):
         if cache_dir is None:
             cache_dir = CACHE_DIR
+        if not os.path.isdir(cache_dir):
+            os.makedirs(cache_dir)
```

Comment:
Might have some drawbacks, e.g. with regard to older Python versions though.

It was just that `mkdir -p` came into my mind when seeing this.

After all, your current code is OK and readable!